crates/iceberg-catalog/src/catalog/tables.rs

```diff
@@ -283,37 +318,60 @@ impl<C: Catalog, A: AuthZHandler, S: SecretStore>
         }
 
         // ------------------- AUTHZ -------------------
-        let include_staged = false;
+        let authorizer = state.v1_state.authz;
+        authorizer
+            .require_warehouse_action(&request_metadata, warehouse_id, &WarehouseAction::CanUse)
+            .await?;
+        let include_staged: bool = false;
         let include_deleted = false;
         let include_active = true;
 
-        let table_id = C::table_ident_to_id(
+        let mut t = C::Transaction::begin_read(state.v1_state.catalog).await?;
+        let table_id = C::table_to_id(
             warehouse_id,
             &table,
             ListFlags {
                 include_active,
                 include_staged,
                 include_deleted,
             },
-            state.v1_state.catalog.clone(),
+            t.transaction(),
         )
-        .await
-        // We can't fail before AuthZ.
-        .ok()
-        .flatten();
+        .await; // We can't fail before AuthZ.
+        let table_id = authorizer
+            .require_table_action(
+                &request_metadata,
+                warehouse_id,
+                table_id,
+                &TableAction::CanGetMetadata,
+            )
+            .await
+            .map_err(set_not_found_status_code)?;
+
+        let (read_access, write_access) = futures::try_join!(
+            authorizer.is_allowed_table_action(
+                &request_metadata,
+                warehouse_id,
+                table_id,
+                &TableAction::CanWriteData,
+            ),
+            authorizer.is_allowed_table_action(
+                &request_metadata,
+                warehouse_id,
+                table_id,
+                &TableAction::CanWriteData,
+            ),
+        )?;
 
-        A::check_load_table(
-            &request_metadata,
-            warehouse_id,
-            Some(&table.namespace),
-            table_id,
-            state.v1_state.auth,
-        )
-        .await?;
+        let storage_permissions = if write_access {
+            Some(StoragePermissions::ReadWrite) // ToDo Tobi: Double check with gc.enabled / deletion enabled flag
```

Comment:
1. don't think we're doing anything with gc.enabled yet
2. we do have ReadWriteDelete as a permission, should we get rid of that variant? All the cloud-providers seem to separate write and delete

Reply:
I think for now we need to grant ReadWriteDelete as expire snapshots happens externally.